src/handlers/Handler.ts

```diff
@@ -450,4 +461,343 @@ export class Handler implements IHandler {
+
+        try {
+            // Show user query first
+            const queryDisplayMessage = `${t(Translations.LLM_USER_QUERY_DISPLAY, this.language, { query })}\n\n🤖 ${t(Translations.LLM_AI_THINKING, this.language)}`;
+            messageBuilder.setText(queryDisplayMessage);
+            await this.read.getNotifier().notifyUser(this.sender, messageBuilder.getMessage());
+
+            // Enhance query with email addresses for mentioned users
+            const usernameService = new UsernameService(this.read);
+            const enhancedQuery = await usernameService.enhanceQueryWithEmails(query);
+
+            // Process enhanced query with LLM
+            const llmService = new LLMService(this.http, this.app.getLogger());
+            const { toolCalls, error } = await llmService.processNaturalLanguageQuery(enhancedQuery);
+
+            // Create new message for results
+            const resultMessageBuilder = this.modify
+                .getCreator()
+                .startMessage()
+                .setSender(appUser)
+                .setRoom(this.room)
+                .setGroupable(false);
+
+            if (error) {
+                resultMessageBuilder.setText(error);
+                return this.read.getNotifier().notifyUser(this.sender, resultMessageBuilder.getMessage());
+            }
+
+            if (toolCalls && toolCalls.length > 0) {
+                const toolCall = toolCalls[0];
+                const toolDisplayName = this.getToolDisplayName(toolCall.function.name);
+                
+                // Parse tool arguments with better error handling
+                let args: any;
+                try {
+                    args = JSON.parse(toolCall.function.arguments);
+                } catch (parseError) {
+                    let fixedArguments = toolCall.function.arguments;
+                    
+                    fixedArguments = fixedArguments.replace(/("content"\s*:\s*")([^"]*)\n([^"]*?)(")/g, (match, start, content1, content2, end) => {
+                        const fixedContent = (content1 + content2).replace(/\n/g, '\\n').replace(/\r/g, '\\r');
+                        return start + fixedContent + end;
+                    });
+                    
+                    fixedArguments = fixedArguments.replace(/("content"\s*:\s*")([^"]*"[^"]*?)(")/g, (match, start, content, end) => {
+                        const fixedContent = content.replace(/"/g, '\\"');
+                        return start + fixedContent + end;
+                    });
+                    
+                    try {
+                        args = JSON.parse(fixedArguments);
+                    } catch (secondParseError) {
+                        resultMessageBuilder.setText(t(Translations.LLM_PARSING_ERROR, this.language));
+                        return this.read.getNotifier().notifyUser(this.sender, resultMessageBuilder.getMessage());
+                    }
+                }
+                
```

Comment:
what is this piece of code doing to many regex no explanation try to give me some example in comment its hard to understand

Reply:
Then why is it still in code ?